tests/dependencies.py

```diff
@@ -0,0 +1,37 @@
+import importlib
+import re
+
+from tests import TestCase
+import sys
+import unittest
+
+
+class TestDependencies(TestCase):
+    """
+    Test of dependencies for specific versions of PyBaMM
+    """
+
+    # Test that optional dependencies are not installed in the core version of PyBaMM
+    def test_optional_dependencies(self):
+        pattern = re.compile(r"^([^>=;\[]+)\b.*$")
+        json_deps = importlib.metadata.metadata("pybamm").json["requires_dist"]
```

Comment:
Could you add an additional test case that checks if the `import pybamm` statement works without any optional dependencies being installed? If `import pybamm` cannot be made to work, we can use `importlib.import_module("pybamm")` as an alternative.

Also, this test seems a bit redundant to me, since `pip install -e .` or `pip install .` will never install optional dependencies and such behaviour is probably being tested in `pip` already.